service/lib/dinstaller/dbus/software_service.rb

```diff
@@ -42,6 +42,7 @@ class SoftwareService
       # @param logger [Logger]
       def initialize(config, logger = nil)
         @backend = DInstaller::Software.new(config, logger)
+        @backend.on_progress_change { dispatch }
```

Comment:
BTW, shouln't logger be initialized before? I mean, I would try to use the very same logger for both, backend and service.